tron/serialize/runstate/dynamodb_state_store.py

```diff
@@ -88,23 +92,36 @@ def _merge_items(self, first_items, remaining_items) -> dict:
         return deserialized_items
 
     def save(self, key_value_pairs) -> None:
-        """
-        Remove all previous data with the same partition key, examine the size of state_data,
-        and splice it into different parts under 400KB with different sort keys,
-        and save them under the same partition key built.
-        """
-        try:
-            for key, val in key_value_pairs:
-                self._delete_item(key)
-                self[key] = pickle.dumps(val)
-        except Exception as e:
-            error = f'tron_dynamodb_save_failure: failed to save items with keys \n{key_value_pairs.keys()}\n to dynamodb\n{repr(e)}'
-            log.error(error)
+        for key, val in key_value_pairs:
+            self.save_queue.append((key,pickle.dumps(val),))
+
+    def _save_loop(self):
+        while True:
+            for _ in range(len(self.save_queue)):
+                try:
+                    key, val = self.save_queue.popleft()
+                    # Remove all previous data with the same partition key
+                    # TODO: only remove excess partitions if new data has fewer
+                    self._delete_item(key)
+                    self[key] = pickle.dumps(val)
+                except Exception as e:
+                    error = (
+                        'tron_dynamodb_save_failure: failed to save key'
+                        f'"{key}" to dynamodb:\n{repr(e)}'
+                    )
+                    log.error(error)
```

Comment:
are you sure this is all we want to do? I'm kind of concerned we'll get into the sort of situation we do in taskproc where threads die (and dump an exception along the way) without us noticing. might it be worth us raising the exception here?

Reply:
well in this case we're not killing the thread, i just rely on aws dynamodb client to continue working i guess, not sure if it's a good assumption. there are multiple reasons we can get exception here - throttling, network timeout, etc, not all of them are terminal. in fact i think i should probably add a retry here for that key/value.

how about i add an error counter and throw a system kill after say 100 errors?